packages/api/src/task/scheduler.ts

```diff
@@ -608,29 +601,41 @@ export class TaskScheduler {
         ),
       );
     }
+    const event = updates.deleted ? "asset.deleted" : "asset.updated";
+    const publishWebhook = async (asset: Asset) => {
+      const snapshot = await toExternalAsset(asset, this.config, true);
+      const timestamp = Date.now();
+      await this.queue.publishWebhook(`events.${event}`, {
+        type: "webhook_event",
+        id: uuid(),
+        timestamp,
+        event,
+        userId: asset.userId,
+        projectId: asset.projectId,
+        payload: {
+          asset: {
+            id: asset.id,
+            snapshot,
+          },
+        },
+      });
+    };
+
+    if (event === "asset.deleted") {
+      await publishWebhook(asset);
+    }
```

Comment:
WDYT of sending this after the `db.asset.update` below (but before we update the `asset` var)? Otherwise, we may send the event then the update fails

Reply:
Make sense! doing it